src/json.cpp

```diff
@@ -99,7 +99,18 @@ JsonObject::JsonObject( JsonIn &j )
     final_separator = jsin->get_ate_separator();
 }
 
-void JsonObject::finish()
+void JsonObject::mark_visited( const std::string &name ) const
+{
+#ifndef CATA_IN_TOOL
+    if( test_mode && report_unvisited_members && !reported_unvisited_members ) {
```

Comment:
This level of checking seems a bit excessive on every insert, it's pessimisizing the case where we do want to do tracking.
The emplace is cheap enough that we should probably just continue doing it unconditionally, hiding it behind a compile time option is plenty.

Reply:
Done